src/internal/PolygonList.zig

```diff
@@ -47,6 +51,268 @@ pub fn prepend(self: *PolygonList, alloc: mem.Allocator, poly: Polygon) mem.Allo
     }
 }
 
+/// Returns true if our polygon list is somewhere within the box starting at
+/// (0,0) with the width box_width and the height box_height. Used to check if
+/// we should actually proceed with drawing.
+pub fn inBox(self: *const PolygonList, scale: f64, box_width: i32, box_height: i32) bool {
+    if (box_width < 1 or box_height < 1) {
```

Comment:
PS: This is reflective of the fact that we will be changing the surface validations to not allow zero-size surfaces.